src/machine/timer.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021, Axel Heider <axelheider@gmx.de>
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#include <config.h>
+
+#ifdef CONFIG_KERNEL_MCS
+
+#include <machine/timer.h>
+
+/* Read the current time from the timer. */
+ticks_t getCurrentTime(void)
+{
+    static uint64_t time_last = 0;
```

Comment:
Shouldn't this be a core local variable instead, in case a platform does not has cross-core synchronized timers?

And maybe make the if-check unlikely.

Reply:
Made the if-check unlikely and tracking things in core-local variables now.